modules/services/MapRouletteService.js

```diff
@@ -0,0 +1,384 @@
+    const url = `${MAPROULETTE_API}/challenge/${task.task.parentId}`;
+    const handleResponse = (data) => {
+      task.instruction = marked.parse(data.instruction) || '';
+      task.details = marked.parse(data.description) || '';
+      this._deletedChallenges[data.id] = data.deleted;
+    };
+
+    return fetch(url)
+      .then(utilFetchResponse)
+      .then(handleResponse)
+      .then(() => task);
+  }
+
+
+  /**
+   * postUpdate
+   * @param   task
+   * @param   callback
+   */
+  postUpdate(task, callback) {
+    const context = this.context;
+    const sidebar = context.systems.ui.sidebar;
+    if (this._cache.inflightPost[task.id]) {
+      console.log('Task update already inflight for task:', task);
+      return callback({ message: 'Issue update already inflight', status: -2 }, task);
+    }
+    const commentUrl = `${MAPROULETTE_API}/task/${task.id}/comment`;
+    const updateTaskUrl = `${MAPROULETTE_API}/task/${task.id}/${task.taskStatus}`;
+    const releaseTaskUrl = `${MAPROULETTE_API}/task/${task.taskId}/release`;
+    const controller = new AbortController();
+    this._cache.inflightPost[task.id] = controller;
+
+    fetch(commentUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'apiKey': task.mapRouletteApiKey
+      },
+      body: JSON.stringify({ actionId: 2, comment: task.comment }),
+      signal: controller.signal
+    })
+    .then(response => {
+      if (!response.ok) throw new Error(`Error posting comment: ${response.statusText}`);
+      return response.json();
+    });
+
+    fetch(updateTaskUrl, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+        'apiKey': task.mapRouletteApiKey
+      },
+      signal: controller.signal
+    })
+    .then(response => {
+      if (!response.ok) throw new Error(`Error updating task: ${response.statusText}`);
+      return response.text();  // get the response text
+    })
+    .then(() => {
+      // Release task
+      return fetch(releaseTaskUrl, {
+        signal: controller.signal,
+        headers: {
+          'apiKey': task.mapRouletteApiKey
+        }
+      });
+    })
+    .then(response => {
+      if (!response.ok) throw new Error(`Error releasing task: ${response.statusText}`);
+      return response.json();
+    })
+    .then(() => {
+      // All requests completed successfully
+      delete this._cache.inflightPost[task.id];
+      this.removeTask(task);
+      sidebar.hide();
+      if (!(task.id in this._cache.closed)) {
+        this._cache.closed[task.id] = 0;
+        if (task.comment) {
+          task.comment += ` #maproulette mpr.lt/c/${task.task.parentId}/t/${task.taskId}`;
+          this._cache.comment[task.id] = { id: task.id, comment: task.comment };
+        }
+      }
+      this._cache.closed[task.id] += 1;
+      if (callback) callback(null, task);
+    })
+    .catch(err => {
+      console.log('In catch block', err);
+      // Handle any errors
```

Comment:
same here?  or `this._cache.inflightPost[task.id]` is mixing up the abort controllers from 2 different fetches which seems bad.  We might be able to just get rid of them, since it's not like the user can abort their clicking of the "I'm done" button.